### PR TITLE
curve-fuse: only update dirty inode metadata

### PR DESCRIPTION
**Pull request by wu-hanqing**
Signed-off-by: Hanqing Wu <wuhanqing@corp.netease.com>

### What problem does this PR solve?

Issue Number: #1845 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by wu-hanqing** [comment]
recheck

**Comment by wu-hanqing** [comment]
recheck

File: curvefs/src/client/inode_wrapper.cpp [hunk 1]
```diff
@@ -336,19 +337,25 @@ CURVEFS_ERROR InodeWrapper::UnLinkLocked(uint64_t parent) {
             auto err = FlushVolumeExtent();
             if (err != CURVEFS_ERROR::OK) {
                 LOG(ERROR) << "Flush volume extent failed, inodeid: "
-                           << inode_.inodeid() << ", error: " << err;
+                        << inode_.inodeid() << ", error: " << err;
```
Comment (SeanHai) [inline curvefs/src/client/inode_wrapper.cpp:340]: output alignment

File: curvefs/src/client/inode_wrapper.cpp [hunk 2]
```diff
@@ -336,19 +337,25 @@ CURVEFS_ERROR InodeWrapper::UnLinkLocked(uint64_t parent) {
...
+        *dirtyAttr_.mutable_parent() = inode_.parent();
+
+        MetaStatusCode ret = metaClient_->UpdateInodeAttr(
+            inode_.fsid(), inode_.inodeid(), dirtyAttr_);
         if (ret != MetaStatusCode::OK) {
             LOG(ERROR) << "metaClient_ UpdateInodeAttr failed"
-                       << ", MetaStatusCode = " << ret
-                       << ", MetaStatusCode_Name = " << MetaStatusCode_Name(ret)
-                       << ", inodeid = " << inode_.inodeid();
+                    << ", MetaStatusCode = " << ret
```
Comment (SeanHai) [inline curvefs/src/client/inode_wrapper.cpp:352]: output alignment

File: curvefs/src/client/inode_wrapper.cpp [hunk 3]
```diff
@@ -358,8 +365,8 @@ CURVEFS_ERROR InodeWrapper::UnLinkLocked(uint64_t parent) {
 
 CURVEFS_ERROR
 InodeWrapper::UpdateInodeStatus(InodeOpenStatusChange statusChange) {
-    MetaStatusCode ret =
-        metaClient_->UpdateInodeAttrWithOutNlink(inode_, statusChange);
+    MetaStatusCode ret = metaClient_->UpdateInodeAttrWithOutNlink(
```
Comment (SeanHai) [inline curvefs/src/client/inode_wrapper.cpp:368]: UpdateInodeStatus this function is not used anymore, you can delete it.

File: curvefs/src/client/inode_wrapper.cpp (hunk 2 above)
Comment (wu-hanqing) [reply curvefs/src/client/inode_wrapper.cpp]: fix

File: curvefs/src/client/inode_wrapper.cpp (hunk 3 above)
Comment (wu-hanqing) [reply curvefs/src/client/inode_wrapper.cpp]: fix

**Comment by wu-hanqing** [comment]
recheck

**Review by SeanHai — APPROVED** [review]

**Review by ilixiaocui — APPROVED** [review]

**Comment by wu-hanqing** [comment]
recheck

**Comment by wu-hanqing** [comment]
recheck

**Comment by YunhuiChen** [comment]
recheck

**Pull request merged**